snuba/datasets/pipeline/single_query_plan_pipeline.py

```diff
@@ -0,0 +1,42 @@
+from snuba.datasets.pipeline.query_pipeline import (
+    QueryPipeline,
+    QueryPipelineBuilder,
+    _execute_query_plan_processors,
+)
+from snuba.datasets.plans.query_plan import (
+    ClickhouseQueryPlan,
+    ClickhouseQueryPlanBuilder,
+    QueryRunner,
+)
+from snuba.request import Request
+from snuba.web import QueryResult
+
+
+class SingleQueryPlanPipeline(QueryPipeline):
+    """
+    A query pipeline for a single query plan.
+    """
+
+    def __init__(
+        self, request: Request, query_plan_builder: ClickhouseQueryPlanBuilder
+    ):
+        self.__request = request
+        self.__query_plan = query_plan_builder.build_plan(request)
+        _execute_query_plan_processors(self.__query_plan, self.__request)
```

Comment:
Why do you perform query processing in the constructor instead of in the execute method ?

Reply:
This should be in execute, thanks